src/System.Private.ServiceModel/src/System/ServiceModel/Channels/RTHttpClientHandler.cs

```diff
@@ -0,0 +1,475 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if FEATURE_NETNATIVE
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Globalization;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Security.Cryptography.X509Certificates;
+using System.ServiceModel.Security;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Windows.Security.Cryptography;
+using Windows.Security.Cryptography.Certificates;
+
+using RTApiInformation = Windows.Foundation.Metadata.ApiInformation;
+using RTHttpBaseProtocolFilter = Windows.Web.Http.Filters.HttpBaseProtocolFilter;
+using RTHttpCacheReadBehavior = Windows.Web.Http.Filters.HttpCacheReadBehavior;
+using RTHttpCacheWriteBehavior = Windows.Web.Http.Filters.HttpCacheWriteBehavior;
+using RTHttpCookieUsageBehavior = Windows.Web.Http.Filters.HttpCookieUsageBehavior;
+using RTPasswordCredential = Windows.Security.Credentials.PasswordCredential;
+
+namespace System.ServiceModel.Channels
+{
+    internal class RTHttpClientHandler : DelegatingHandler
+    {
+        private static readonly Lazy<bool> s_RTCookieUsageBehaviorSupported =
+            new Lazy<bool>(InitRTCookieUsageBehaviorSupported);
+
+        //#region Fields
+
+        private readonly RTHttpBaseProtocolFilter _rtFilter;
+        private readonly RTHttpHandlerToFilter _handlerToFilter;
+
+        private volatile bool _operationStarted;
+        private volatile bool _disposed;
+
+        private CookieContainer _cookieContainer;
+        private bool _useCookies;
+        private DecompressionMethods _automaticDecompression;
+        private X509Certificate2Collection _clientCertificates;
+
+
+        //#endregion Fields
+
+        //#region Properties
+
+        public bool UseCookies
+        {
+            get { return _useCookies; }
+            set
+            {
+                CheckDisposedOrStarted();
+                _useCookies = value;
+            }
+        }
+
+        public CookieContainer CookieContainer
+        {
+            get { return _cookieContainer; }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException("value");
+                }
+                if (!UseCookies)
+                {
+                    throw new InvalidOperationException(string.Format(CultureInfo.InvariantCulture,
+                        "net_http_invalid_enable_first", "UseCookies", "true"));
```

Comment:
Questions: why doesn't this use SR.Format, and will it produce a meaningful error?